jetty-server/src/main/java/org/eclipse/jetty/server/HttpConnection.java

```diff
@@ -60,8 +60,7 @@
     private final HttpGenerator _generator;
     private final HttpChannelOverHttp _channel;
     private final HttpParser _parser;
-    private final AtomicInteger _contentBufferReferences = new AtomicInteger();
-    private volatile ByteBuffer _requestBuffer = null;
+    private volatile RetainableByteBuffer _retainableByteBuffer;
```

Comment:
I know this is a draft... but thought I'd lob in some comments anyway....

having a volatile reference to a class that itself holds a reference to a buffer is a bit too much. I'd like to see:

```suggestion
    private final RetainableByteBuffer _retainableByteBuffer;
```

which means the `RetainableByteBuffer` needs some modifications... but it will anyway if it is to end up being the pooled instance

Reply:
Sounds like a case of premature optimization :smiley:. Let's first see how the `RetainableByteBuffer` pool API pans out.

Reply:
So far, my perf tests tend to show that any loss due to the double indirection is largely compensated by the much more efficient pooling algorithm that this change allows.